cpp/src/reductions/simple.cuh

```diff
@@ -104,9 +104,15 @@ std::unique_ptr<scalar> fixed_point_reduction(column_view const& col,
     }
   }();
 
-  auto const scale = std::is_same<Op, cudf::reduction::op::product>::value
-                       ? numeric::scale_type{col.type().scale() * (col.size() - col.null_count())}
-                       : numeric::scale_type{col.type().scale()};
+  auto const scale = [&] {
+    if (std::is_same<Op, cudf::reduction::op::product>::value)
+      return numeric::scale_type{col.type().scale() * (col.size() - col.null_count())};
```

Comment:
I have been experimenting changing the underlying `size_type` and it caused a compile error (warning maybe) on this line since `scale_type` is `int32`. Changing this line (and the others to match) to use `static_cast` instead fixed the error.
```suggestion
      return static_cast<numeric::scale_type>(col.type().scale() * (col.size() - col.null_count()));
```